### ISSUE TITLE
Add --disable-smooth-scrolling command line switch.

### ISSUE THREAD

**Issue by akashnimare** [electron/electron#11583]
* Electron version: 1.7.9
* Operating system: macOS

This is a feature request. I feel that Chromium's smooth scrolling feature is annoying. It would be nice if that could be turned off by some command line switch. Currently, this command line switch is not supported by electron

**Comment by codebytere** [comment]
With the volume of issues we receive, we'll need you to fill out the issue template before we can consider this feature request.

**Comment by sethlu** [comment]
@akashnimare by saying smooth scrolling, do you mean that the page continues scrolling even though the user has stepped away from the interaction?

As a little bit of extra context: `#smooth-scrolling` exists in Chrome as a flag on Windows, Linux, Chrome OS, Android to opt in smooth animation in page content scrolling.

**Comment by akashnimare** [comment]
@sethlu please see https://github.com/zulip/zulip-electron/issues/374. @Be-ing do you mind adding few details here about the scrolling issue on the electron app?

**Comment by Be-ing** [comment]
@akashnimare, sorry for the delay. I don't know why I didn't see your ping on this issue. I find this smooth scrolling really annoying because it actually makes scrolling really laggy. The longer the distance scrolled, the longer the UI lags.

I'm not sure that a command line switch is the best way to implement this. I would rather control it from the GUI, but AFAIK there is no way to do that without every Electron app specifically implementing that itself.

**Comment by akashnimare** [comment]
@Be-ing we can add this in GUI if electron supports following api - 
```
app.commandLine.appendSwitch('disable-smooth-scrolling', 'true')
```
@sethlu, @codebytere does this api work in electron v1.8.2?

**Comment by sethlu** [comment]
@akashnimare It turns out scrolling inertia is a macOS a11y feature.

> As a little bit of extra context: `#smooth-scrolling` exists in Chrome as a flag on Windows, Linux, Chrome OS, Android to opt in smooth animation in page content scrolling.

Since there isn't a native smooth scrolling effect on those OS'es, `#smooth-scrolling` is an opt-in feature to emulate this user interaction; however, it's not yet possible to opt-out on macOS easily with a public system API.

Here's an alternative attempt though: https://github.com/d4nyll/lethargy

**Comment by Be-ing** [comment]
> Since there isn't a native smooth scrolling effect on those OS'es, #smooth-scrolling is an opt-in feature to emulate this user interaction

Huh? It seems to be opt-out on Linux at least. If it was opt-in, or not implemented at all, I wouldn't care. But I experience smooth scrolling as annoying and want to turn it off. I can do this in Firefox and Chromium, so I'm not sure why I can't do it in Electron apps.

> however, it's not yet possible to opt-out on macOS easily with a public system API

What is the problem with that? If the feature doesn't work on macOS because it is overridden by the system setting, then don't make it do anything on macOS. IMO it doesn't make sense to not have this option on other platforms because it is overridden by a system setting on one platform.

> Here's an alternative attempt though: https://github.com/d4nyll/lethargy

Are you suggesting that every Electron app should use a library to solve this on their own?

**Comment by sethlu** [comment]
@Be-ing Right, sorry I closed this earlier since I thought we were addressing a macOS specific issue. Let me reopen the issue and someone with more experience in Linux can help with this.

**Comment by sethlu** [comment]
Some thoughts on opting out momentum

https://developer.apple.com/library/archive/releasenotes/DriversKernelHardware/RN-MagicMouse/

**Comment by welkie** [comment]
@sethlu They seem to have closed that issue claiming it isn't possible. This is a big problem. Many people experience smooth scrolling as a negative (myself included). The only app I use that I believe uses Electron that has this issue is Slack. But I use Slack enough during my work day to be annoyed by it.

Notably, these apps which I believe use Electron don't have this problem:

* Visual Studio Code
* Telegram

That leads me to believe there must be something the devs are doing to turn it off, and it would be nice to make it so devs can easily expose an option to users to turn it off.

**Comment by welkie** [comment]
I read a bit more and now I understand the role the `--disable-smooth-scrolling` switch plays. We use it to start the Electron app (ex. `slack --disable-smooth-scrolling`) and it works fine since the Electron app ends up passing the configuration to the Chromium process it starts. This solution works for me when it comes to disabling smooth scrolling in Slack.

The only friction at this point is that I need to learn more about my OS (Ubuntu 18.10) to find out how to make it so when I use the DE shortcut to launch Slack, it includes that configuration. I do think it would be better UX if there were a setting in Slack's preferences screen to toggle this and Slack could store this setting the same way it stores other settings. I bet this setting change would require restarting Slack since it needs a new Chromium process to be started, but as a Slack user, I'd be perfectly ok with being prompted to restart Slack when I change the setting.

**Comment by solarkraft** [comment]
Indeed the command line switch works (oh, how good it feels now that the sponginess is removed!), but running to each app maintainer to add a switch seems a bit impractical. 
#28537 would allow for system wide configuration.

**Comment by DubbaThony** [comment]
Not only matter of preference, there is also mouse bug in linux guests under vmware that makes  scrolling while moving mouse impossible without this flag. Adding the flag is workaround, not proper fix, but AFAIK chromium is not planning the issue fix. Its something with how libinput handles tablet input and scrolling.

If the feature would be implemented in electron globally, it would allow workaround on all electron-based apps, which would be major improvement for people who work within VMs.

**Comment by MarshallOfSound** [comment]
`--disable-smooth-scrolling` works (see the comments above), a system wide toggle is a different ask (#46473).
